bgpd/bgp_pbr.c

```diff
@@ -459,6 +504,373 @@ void bgp_pbr_print_policy_route(struct bgp_pbr_entry_main *api)
 	zlog_info("%s", return_string);
 }
 
+static void bgp_pbr_flush_entry(struct bgp *bgp, struct bgp_pbr_action *bpa,
+				struct bgp_pbr_match *bpm,
+				struct bgp_pbr_match_entry *bpme)
+{
+	/* if bpme is null, bpm is also null
+	 */
+	if (bpme == NULL)
+		return;
+	/* ipset del entry */
+	if (bpme->installed) {
+		bgp_send_pbr_ipset_entry_match(bpme, false);
+		bpme->installed = false;
+		bpme->backpointer = NULL;
+	}
+	hash_release(bpm->entry_hash, bpme);
+	if (hashcount(bpm->entry_hash) == 0) {
+		/* delete iptable entry first */
+		/* then delete ipset match */
+		if (bpm->installed) {
+			if (bpm->installed2) {
```

Comment:
what the heck is installed2?  Let's find a better name for the variable.

Reply:
installed_in_iptable is the real goal of that variable.